crates/rspack_core/src/cache/occasion/resolve_module.rs

```diff
@@ -68,7 +68,9 @@ impl ResolveModuleOccasion {
     let data = generator(args).await?;
     let mut paths = Vec::new();
     if let ResolveResult::Resource(resource) = &data {
-      paths.push(resource.path.as_path());
+      if resource.path.is_file() {
+        paths.push(resource.path.as_path());
```

Comment:
I don't think it is a correct fix, because the error was caused by missed calculating the hash for directory at snapshot. 
The ` fn create_snapshot<F>(&self, paths: &[&Path], f: F)` should be ` fn create_snapshot<F>(&self, paths: &[&Path], dirs: &[&Dir] f: F)`, can you fix the function and update dirs hash logic?

Reply:
I use the path in the directory for hash computation, do you think this is OK?

Reply:
Seems like just setting the hash to the `"directory"` string like webpack is enough
https://github.com/webpack/webpack/blob/5b1fd3ebbaf0fd5b7706c29f0be12b7b0150e743/lib/FileSystemInfo.js#L2907

Reply:
Thank you. I have modified the code according to this